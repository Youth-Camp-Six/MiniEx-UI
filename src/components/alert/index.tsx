import Alert from './alert';

export * from './type';
export { Alert };