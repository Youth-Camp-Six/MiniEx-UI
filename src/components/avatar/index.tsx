import Avatar from './avatar';

export * from './type';
export { Avatar };
