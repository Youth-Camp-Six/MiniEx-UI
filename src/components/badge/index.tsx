import Badge from './badge';

export * from './type';
export { Badge };
