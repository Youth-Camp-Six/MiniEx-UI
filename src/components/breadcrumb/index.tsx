import Breadcrumb from './breadcrumb';

export * from './type';
export { Breadcrumb };
