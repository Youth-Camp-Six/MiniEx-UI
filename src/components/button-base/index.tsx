import ButtonBase from './button-base';

export { ButtonBase };
