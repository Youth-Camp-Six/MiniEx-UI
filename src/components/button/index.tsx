import Button from './button';

export * from './type';
export { Button };
