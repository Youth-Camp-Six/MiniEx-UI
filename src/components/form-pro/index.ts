import FormPro from './formpro';
import Field from './field';
import useForm from './use-form';

export { Field, useForm, FormPro };
