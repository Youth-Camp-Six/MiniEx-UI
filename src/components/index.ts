import '../style/index.less';

export * from './button';
export * from './menu';
export * from './title';
export * from './flex';
export * from './icon/icon';
export * from './tabs';
export * from './toast';
export * from './breadcrumb';
export * from './dropdown';
export * from './input';
export * from './input-tag';
export * from './form';
export * from './checkbox';
export * from './radio';
export * from './switch';
export * from './avatar';
export * from './badge';
export * from './tag';
export * from './alert';
export * from './popover';
export * from './select';
export * from './upload';
