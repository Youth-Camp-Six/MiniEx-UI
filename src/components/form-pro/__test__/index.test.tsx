/**
 * @jest-environment jsdom
 */

import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { Field, FormPro, useForm } from '..';
import { Button, CheckboxGroup, RadioGroup } from '../../../../src/components';

const radioOptions = [
  { label: 'Unknown', value: '0' },
  { label: 'man', value: '1' },
  { label: 'woman', value: '2' },
];

const checkboxOptions = [
  { label: 'labe-0', value: 0 },
  { label: 'labe-1', value: 1 },
  { label: 'labe-2', value: 2 },
];

describe('test Form comonent', () => {
  it('Form snapshot test', () => {
    const TestComponent = () => {
      const [form] = useForm();

      return (
        <FormPro labelwidth='100px' form={form}>
          <Field name='sex' label='性别'>
            <RadioGroup options={radioOptions}></RadioGroup>
          </Field>
          <Field name='checkboxData' label='lebel'>
            <CheckboxGroup options={checkboxOptions}></CheckboxGroup>
          </Field>
          <div>
            <Button>提交</Button>
          </div>
        </FormPro>
      );
    };

    const wrapper = render(<TestComponent />);

    expect(wrapper.container.firstChild).toMatchSnapshot();
  });

  it('submits the form and calls onFinishHandler', () => {
    const onFinishHandler = jest.fn();
    const onFinishFailedHandler = jest.fn();

    const { getByLabelText } = render(
      <FormPro onFinish={onFinishHandler} onFinishFailed={onFinishFailedHandler}>
        <Field name='sex' label='性别'>
          <RadioGroup
            options={[
              { label: 'Unknown', value: '0' },
              { label: 'man', value: '1' },
              { label: 'woman', value: '2' },
            ]}
          />
        </Field>
        <Field name='checkboxData' label='lebel'>
          <CheckboxGroup
            options={[
              { label: 'labe-0', value: 0 },
              { label: 'labe-1', value: 1 },
              { label: 'labe-2', value: 2 },
            ]}
          />
        </Field>
        <Field>
          <Button>提交</Button>
        </Field>
      </FormPro>
    );

    // Select radio button
    fireEvent.click(getByLabelText('man'));

    // Select first and second checkbox
    fireEvent.click(getByLabelText('labe-0'));
    fireEvent.click(getByLabelText('labe-1'));
  });
});
