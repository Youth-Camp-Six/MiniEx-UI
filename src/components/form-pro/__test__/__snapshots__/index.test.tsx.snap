// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Form comonent Form snapshot test 1`] = `
<form
  class="mi-form"
>
  <div
    class="mi-form-item mi-form-item-x mi-form-item-end"
  >
    <div
      class="mi-form-item-label"
      style="width: 100px;"
    >
      性别
    </div>
    <div
      class="mi-form-item-view"
    >
      <div
        class="mi-form-item-view-inner"
      >
        <div
          class="mi-flex mi-radio lew-radio-group-x lew-radio-group-medium mi-flex-x mi-flex-x-start mi-flex-y-center"
          style="gap: 15px;"
        >
          <label
            class="mi-radio 
    
    mi-radio-medium
    
    "
            data-testid="radio-0"
          >
            <div
              class="icon-radio-box"
            >
              <div
                class="icon-radio"
              />
            </div>
            <input
              style="display: none;"
              type="radio"
              value=""
            />
            <span
              class="mi-radio-label"
            >
               
              Unknown
            </span>
          </label>
          <label
            class="mi-radio 
    
    mi-radio-medium
    
    "
            data-testid="radio-1"
          >
            <div
              class="icon-radio-box"
            >
              <div
                class="icon-radio"
              />
            </div>
            <input
              style="display: none;"
              type="radio"
              value=""
            />
            <span
              class="mi-radio-label"
            >
               
              man
            </span>
          </label>
          <label
            class="mi-radio 
    
    mi-radio-medium
    
    "
            data-testid="radio-2"
          >
            <div
              class="icon-radio-box"
            >
              <div
                class="icon-radio"
              />
            </div>
            <input
              style="display: none;"
              type="radio"
              value=""
            />
            <span
              class="mi-radio-label"
            >
               
              woman
            </span>
          </label>
        </div>
      </div>
      <div
        class="mi-form-item-view-message"
      />
    </div>
  </div>
  <div
    class="mi-form-item mi-form-item-x mi-form-item-end"
  >
    <div
      class="mi-form-item-label"
      style="width: 100px;"
    >
      lebel
    </div>
    <div
      class="mi-form-item-view"
    >
      <div
        class="mi-form-item-view-inner"
      >
        <div
          class="mi-flex mi-checkbox mi-checkbox-group-x mi-checkbox-group-medium mi-flex-x mi-flex-x-start mi-flex-y-center"
          style="gap: 15px;"
        >
          <label
            class="mi-checkbox 
        
        
        
        mi-checkbox-medium
        
        "
            data-testid="checkbox-0"
          >
            <div
              class="icon-checkbox-box"
            >
              <svg
                class="icon-checkbox"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="4"
                viewBox="0 0 24 24"
                width="24"
              >
                <polyline
                  points="20 6 9 17 4 12"
                />
              </svg>
            </div>
            <input
              style="display: none;"
              type="checkbox"
            />
            <span
              class="mi-checkbox-label"
            >
              labe-0
            </span>
          </label>
          <label
            class="mi-checkbox 
        
        
        
        mi-checkbox-medium
        
        "
            data-testid="checkbox-1"
          >
            <div
              class="icon-checkbox-box"
            >
              <svg
                class="icon-checkbox"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="4"
                viewBox="0 0 24 24"
                width="24"
              >
                <polyline
                  points="20 6 9 17 4 12"
                />
              </svg>
            </div>
            <input
              style="display: none;"
              type="checkbox"
            />
            <span
              class="mi-checkbox-label"
            >
              labe-1
            </span>
          </label>
          <label
            class="mi-checkbox 
        
        
        
        mi-checkbox-medium
        
        "
            data-testid="checkbox-2"
          >
            <div
              class="icon-checkbox-box"
            >
              <svg
                class="icon-checkbox"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="4"
                viewBox="0 0 24 24"
                width="24"
              >
                <polyline
                  points="20 6 9 17 4 12"
                />
              </svg>
            </div>
            <input
              style="display: none;"
              type="checkbox"
            />
            <span
              class="mi-checkbox-label"
            >
              labe-2
            </span>
          </label>
        </div>
      </div>
      <div
        class="mi-form-item-view-message"
      />
    </div>
  </div>
  <div>
    <button
      class="btn btn-default"
    >
      提交
    </button>
  </div>
</form>
`;
